src/lido/infos.ts

```diff
@@ -0,0 +1,278 @@
+import { PublicKey, Connection } from "@solana/web3.js";
+import { TOKEN_PROGRAM_ID, ASSOCIATED_TOKEN_PROGRAM_ID, AccountLayout } from "@solana/spl-token-v2";
+import axios from "axios";
+import BN from "bn.js";
+import { IInstanceVault, IVaultInfoWrapper } from "../types";
+import { LIDO_ADDRESS, LIDO_PROGRAM_ID, ST_SOL_MINT_ADDRESS } from "./ids";
+import {
+  LIDO_LAYOUT_V1,
+  LIDO_LAYOUT_V2,
+  LIDO_VERSION_CHECK_LAYOUT,
+  MAINTAINER_LIST_ACCOUNT_LAYOUT,
+  VALIDATOR_LIST_ACCOUNT_LAYOUT,
+} from "./layout";
+import * as types from ".";
+
+let infos: IInstanceVault;
+
+infos = class InstanceLido {
+  static async getAllVaults(connection: Connection): Promise<types.VaultInfo[]> {
+    const vault = await this.getVault(connection, LIDO_ADDRESS);
+
+    return [vault];
+  }
+
+  static async getAllVaultWrappers(connection: Connection): Promise<VaultInfoWrapper[]> {
+    return (await this.getAllVaults(connection)).map((vault) => new VaultInfoWrapper(vault));
+  }
+
+  static async getVault(connection: Connection, vaultId: PublicKey): Promise<types.VaultInfo> {
+    if (!vaultId.equals(LIDO_ADDRESS)) throw Error(`Error: Lido vaultId must match ${LIDO_ADDRESS.toBase58()}`);
+
+    const vaultAccountInfo = await connection.getAccountInfo(vaultId);
+    if (!vaultAccountInfo) throw Error("Error: Cannot get solido account");
+
+    let vault: types.VaultInfo = this.parseVault(vaultAccountInfo.data, vaultId);
+
+    // Fetch validators if using Solido v2
+    if (vault.validatorList) {
+      const validatorListAccount = await connection.getAccountInfo(vault.validatorList);
+      if (!validatorListAccount) throw Error("Error: Cannot get validator list account");
+
+      vault.validators = this._parseValidatorListAccount(validatorListAccount.data);
+    }
+
+    // Fetch maintainers if using Solido v2
+    if (vault.maintainerList) {
+      const maintainerListAccount = await connection.getAccountInfo(vault.maintainerList);
+      if (!maintainerListAccount) throw Error("Error: Cannot get maintainer list account");
+
+      vault.maintainers = this._parseMaintainerListAccount(maintainerListAccount.data);
+    }
```

Comment:
Can fetch `validatorList` and `maintainerList` together with `uitls.getMulipleAccounts()` in `@dappio-wonderland/navigator` by importing `utils`, since they might have same condition corresponding to version, so that we can reduce rpc request.